tsdb/importer/import.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2019 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package importer
+
+import (
+	"github.com/prometheus/prometheus/pkg/value"
+	"github.com/prometheus/prometheus/tsdb/fileutil"
+	"io/ioutil"
+	"math"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/go-kit/kit/log"
+	prom_labels "github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/textparse"
+	"github.com/prometheus/prometheus/tsdb"
+	tsdb_labels "github.com/prometheus/prometheus/tsdb/labels"
+)
+
+// Implementing the error interface to create a
+// constant, which cannot be overridden.
+// https://dave.cheney.net/2016/04/07/constant-errors
+type Error string
+
+func (e Error) Error() string {
+	return string(e)
+}
+
+// This error is thrown when we try to merge/add blocks to an existing TSDB instance,
+// and the new blocks have a time overlap with the current blocks.
+const OverlappingBlocksError = Error("blocks overlap with blocks currently in DB")
+
+type blockTimestampPair struct {
+	start, end int64
+}
+
+// Content Type for the Open Metrics Parser.
+// Needed to init the text parser.
```

Comment:
The flags say the Prometheus format. Which is it?

Reply:
Good catch, I'll update the flags' description. It should be the Open Metrics parser.